tests/filtered_relation/tests.py

```diff
@@ -629,6 +634,15 @@ def test_condition_outside_relation_name(self):
                 ),
             )
 
+    def test_condition_with_exact_lookup_outside_relation_name(self):
+        qs = Author.objects.annotate(
+            book_editor=FilteredRelation(
+                "book__editor",
+                condition=Q(book__author__name="book"),
+            ),
+        ).filter(book_editor__isnull=True)
+        self.assertEqual(qs.count(), 4)
```

Comment:
This test and `test_conditional_expression_with_func` work without this patch so I moved them to a separate PR. Both are also quite surprising especially since `test_condition_outside_relation_name` raises an exception, but this is a discussion for a separate ticket.

Reply:
Check out #17100.